File: src/IO_24/_02_Бондаренко_Тарас_Андрійович/lab5/Word.java
package IO_24._02_Бондаренко_Тарас_Андрійович.lab5;

public class Word implements SentenceElement {
    private final Letter[] letters;

    public Word(String word) {
        letters = new Letter[word.length()];
        for (int i = 0; i < word.length(); i++) {
            letters[i] = new Letter(word.charAt(i));
        }
    }

    @Override
    public String toString() {
        StringBuilder out = new StringBuilder();
        for (Letter let : letters) {
            out.append(let.toString());
        }
        return out.toString();
    }
}

File: Java/lab_5/Word.kt
package OOP.Java.lab_5

/**
 * A class representing a single word is a sentence.
 *
 * @property letters an array of [Letter] objects that make up a word.
 *
 * @constructor a primary constructor accepts an array of [Letter] objects, a secondary one accepts a string representing the entire word.
 */
class Word(var letters: Array<Letter>) {

    override fun toString(): String {
        val wordString = StringBuilder()
        this.letters.forEach { wordString.append(it) }

        return wordString.toString()
    }

    constructor(
        word: String
    ) : this((word.toCharArray().map { letter -> Letter(letter) }).toTypedArray())
}